kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java

```
@@ -0,0 +1,163 @@
+
+        dataSourceDef = new DataSourceDef();
+        dataSourceDef.setUuid( "uuid" );
+        dataSourceDef.setName( "dataSourceName" );
+        dataSourceDef.setConnectionURL( "connectionURL" );
+        dataSourceDef.setUser( "user" );
+        dataSourceDef.setPassword( "password" );
+
+        dataSourceDefEditorContent = new DataSourceDefEditorContent();
+        dataSourceDefEditorContent.setDef( dataSourceDef );
+        dataSourceDefEditorContent.setProject( project );
+
+        originalDataSourceDef = new DataSourceDef();
+        originalDataSourceDef.setUuid( "uuid" );
+        originalDataSourceDef.setName( "dataSourceNameOriginal" );
+        originalDataSourceDef.setConnectionURL( "connectionURLOriginal" );
+        originalDataSourceDef.setUser( "userOriginal" );
+        originalDataSourceDef.setPassword( "passwordOriginal" );
+    }
+
+    @Override
+    protected DefEditorContent getExpectedContent() {
+        return dataSourceDefEditorContent;
+    }
+
+    @Override
+    protected String getExpectedDefString() {
+        return DataSourceDefSerializer.serialize( dataSourceDef );
+    }
+
+    @Override
+    protected String getExpectedFileName() {
+        return dataSourceDef.getName() + ".datasource";
+    }
+
+    @Override
+    protected Def getExpectedDef() {
+        return dataSourceDef;
+    }
+
+    @Override
+    protected Def getOriginalDef() {
+        return originalDataSourceDef;
+    }
+
+    @Override
+    protected String getOriginalDefString() {
+        return DataSourceDefSerializer.serialize( originalDataSourceDef );
+    }
+
+    @Override
+    protected void verifyCreateConditions( boolean global ) {
+        //we wants that:
+        try {
+            // 2) the definition was deployed
+            verify( runtimeManager, times( 1 ) ).deployDataSource( dataSourceDef, DeploymentOptions.create() );
+        } catch ( Exception e ) {
+        }
```

Do you want to fail the test in the event of an exception? The ```runtimeManager.deployDayaSource(...)``` *could* be called - and hence pass the assertion, but generate an error that would be silently ignored.